django/core/handlers/wsgi.py

```diff
@@ -147,18 +146,17 @@ def _get_files(self):
 
 
 class WSGIHandler(base.BaseHandler):
-    initLock = Lock()
     request_class = WSGIRequest
 
-    def __call__(self, environ, start_response):
-        # Set up middleware if needed. We couldn't do this earlier, because
-        # settings weren't available.
-        if self._request_middleware is None:
-            with self.initLock:
-                # Check that middleware is still uninitialized.
-                if self._request_middleware is None:
-                    self.load_middleware()
+    def __init__(self, *args, **kwargs):
+        super(WSGIHandler, self).__init__(*args, **kwargs)
+        # Load middleware on initialization rather than on first request. In
```

Comment:
The "rather than on first request" could be a bit confusing for future readers since it implies they know the history of how things used to work. I think you could probably simplify or drop the comment.